core/LoadStoreInstInfo.hpp

```diff
@@ -127,14 +134,44 @@ namespace olympia
 
         friend bool operator<(const LoadStoreInstInfoPtr & lhs, const LoadStoreInstInfoPtr & rhs)
         {
-            return lhs->getInstUniqueID() < rhs->getInstUniqueID();
+            if(lhs->getInstUniqueID() == rhs->getInstUniqueID())
+            {
+                // if UID is the same, check Uops for vector
+                return lhs->getInstUOpID() < rhs->getInstUOpID();
+            }
+            else
+            {
+                return lhs->getInstUniqueID() < rhs->getInstUniqueID();
+            }
+        }
+
+        void setVectorIter(uint32_t vec_iter){
+            // set number of iterations of VLSU until all bits are loaded into vector register
+            vector_iterations_ = vec_iter;
+        }
+
+        // return current vector iterations
+        uint32_t getVectorIter() const { return vector_iterations_; }
+
+        void setTotalVectorIter(uint32_t total_vec_iter){
+            // set number of iterations of VLSU until all bits are loaded into vector register
+            total_vector_iterations_ = total_vec_iter;
         }
 
+        // return current vector iterations
+        uint32_t getTotalVectorIter() const { return total_vector_iterations_; }
+
+        void setVLSUStatusState(Inst::Status vlsu_status_state ){ vlsu_status_state_ = vlsu_status_state; }
+        Inst::Status getVLSUStatusState(){ return vlsu_status_state_; } 
       private:
         MemoryAccessInfoPtr mem_access_info_ptr_;
         sparta::State<IssuePriority> rank_;
         sparta::State<IssueState> state_;
         bool in_ready_queue_;
+        uint32_t vector_iterations_ = 0;
```

Comment:
Is this really "completed vector iterations"? Could be renamed to be more clear.